tests/e2e/common.go

```diff
@@ -156,32 +157,122 @@ func deleteVeleroCR(client dynamic.Interface, instanceName string, namespace str
 	return veleroClient.Delete(context.Background(), instanceName, metav1.DeleteOptions{})
 }
 
+// Keeping it for now
+func isNamespaceExists(namespace string) error {
+	kubeConf := getKubeConfig()
+	// create client for pod
+	clientset, err := kubernetes.NewForConfig(kubeConf)
+	if err != nil {
+		return err
+	}
+	_, exists := clientset.CoreV1().Namespaces().Get(context.Background(), namespace, metav1.GetOptions{})
+	return exists
+}
+
 // Keeping it for now.
 func isNamespaceDeleted(namespace string) wait.ConditionFunc {
+	fmt.Println("Checking test namespace has been deleted...")
 	return func() (bool, error) {
 		kubeConf := getKubeConfig()
-
-		// create client for pod
 		clientset, err := kubernetes.NewForConfig(kubeConf)
 		if err != nil {
-			return true, err
+			return false, err
 		}
 		_, exists := clientset.CoreV1().Namespaces().Get(context.Background(), namespace, metav1.GetOptions{})
-		if exists == nil {
-			return false, exists
+		if exists != nil {
+			fmt.Println("Test namespace has been deleted")
+			return true, nil
 		}
-		return true, exists
+		return false, exists
 	}
 }
 
-// Keeping it for now
-func isNamespaceExists(namespace string) error {
-	kubeConf := getKubeConfig()
-	// create client for pod
-	clientset, err := kubernetes.NewForConfig(kubeConf)
+func isVeleroDeleted(namespace string, instanceName string) wait.ConditionFunc {
+	fmt.Println("Checking the Velero CR has been deleted...")
+	return func() (bool, error) {
+		kubeConf := getKubeConfig()
```

Comment:
NIT: consider wrapping this into a common function since these lines of code are used in a bunch of places. I.e.:
```
func getDynamicClient() (dynamic.Client, error) {
  kubeConf := getKubeConfig()
  return dynamic.NewForConfig(kubeConf)
}
```